src/Paysera/WalletApi/Listener/PayseraUserIpSetter.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+class Paysera_WalletApi_Listener_PayseraUserIpSetter implements Paysera_WalletApi_EventDispatcher_EventSubscriberInterface
+{
+    /**
+     * @var array
+     */
+    protected $parameters;
+
+    /**
+     * @param array $parameters
+     */
+    public function __construct($parameters)
+    {
+        $this->parameters = $parameters;
+    }
+
+    public function onBeforeRequest(Paysera_WalletApi_Event_RequestEvent $event)
+    {
+        if (isset($this->parameters['Paysera-User-Ip'])) {
```

Comment:
Should we use snake case instead? It could look inconsistent when configuring, because this scope is not headers – it's parameters.

Reply:
I guess that your solution will cause confuse for clients integrating this lib using our documentation. Paysera-User-Ip will be documented as header. We do not document lib behavior.

Reply:
That's another problem, because out of documentation there would be no possible way to know where to put that `Paysera-User-Ip` either :)

Reply:
Maybe it would be clearer to make `AppendHeadersListener` and configure `'headers' => ['Paysera-User-Ip' => $ip]`?

Reply:
Your solution with `AppendHeadersListener` is more reusable and will fit your needs to.